### PR TITLE
[TT-11405] Updating JSON tags and field names for TLS max and min versions

### PR DESCRIPTION
**Pull request by mativm02**
## Description
Updating JSON tags and field names for TLS max and min versions. 

https://tyktech.atlassian.net/browse/TT-11405

## Related Issue
https://tyktech.slack.com/archives/C02AS4BP5HN/p1709041100516449

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning why it's required
- [ ] I would like a code coverage CI quality gate exception and have explained why

___

## **Type**
enhancement

___

## **Description**
- Updated JSON tags and field names related to TLS version configuration across multiple files to improve clarity and consistency.
- Renamed `MinVersion` and `MaxVersion` to `TLSMinVersion` and `TLSMaxVersion` in `HttpServerOptionsConfig` for better readability.
- Adjusted references in the TLS configuration setup in various parts of the codebase to align with the new field names.
- Ensured TLS version consistency and updated the default TLS max version to `VersionTLS12` in the gateway server configuration.

___

## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>config.go</strong><dd><code>Update TLS version field names and JSON tags</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

config/config.go
<li>Updated JSON tags for <code>MaxVersion</code> and <code>MinVersion</code> to <code>tls_max_version</code> and <br><code>tls_min_version</code> respectively in <code>StorageOptionsConf</code>.<br> <li> Renamed fields <code>MinVersion</code> and <code>MaxVersion</code> to <code>TLSMinVersion</code> and <br><code>TLSMaxVersion</code> in <code>HttpServerOptionsConfig</code>.

</details>
    

  </td>
  <td><a href="https:/TykTechnologies/tyk/pull/6078/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>dashboard_register.go</strong><dd><code>Align TLS version configuration references</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/dashboard_register.go
<li>Updated references from <code>MinVersion</code> and <code>MaxVersion</code> to <code>TLSMinVersion</code> and <br><code>TLSMaxVersion</code> in the TLS configuration.

</details>
    

  </td>
  <td><a href="https:/TykTechnologies/tyk/pull/6078/files#diff-f504c88b3d2fa3b56b74c252aab41a934156879ef1150d33714225749e6cc94c">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>proxy_muxer.go</strong><dd><code>Update TLS version configuration in proxy muxer</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/proxy_muxer.go
<li>Adjusted TLS configuration to use the new <code>TLSMinVersion</code> and <br><code>TLSMaxVersion</code> fields.

</details>
    

  </td>
  <td><a href="https:/TykTechnologies/tyk/pull/6078/files#diff-89fb6731880400cb95ba8860c935a308de5f55aaa41aa2c76abf3ee4773d7a87">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rpc_storage_handler.go</strong><dd><code>Update RPC storage handler TLS version configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/rpc_storage_handler.go
<li>Updated RPC configuration to use <code>TLSMinVersion</code> and <code>TLSMaxVersion</code> for <br>SSL version control.

</details>
    

  </td>
  <td><a href="https:/TykTechnologies/tyk/pull/6078/files#diff-8875f75b602664c44b62b67a4da41d748124ad270573a44db4ec977ee5d68021">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>server.go</strong><dd><code>Enforce TLS version consistency and update defaults</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/server.go
<li>Ensured TLS version consistency and updated the default TLS max <br>version to <code>VersionTLS12</code>.<br> <li> Updated checks and assignments to use <code>TLSMinVersion</code> and <code>TLSMaxVersion</code>.

</details>
    

  </td>
  <td><a href="https:/TykTechnologies/tyk/pull/6078/files#diff-4652d1bf175a0be8f5e61ef7177c9666f23e077d8626b73ac9d13358fa8b525b">+4/-4</a>&nbsp; &nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -154,11 +154,11 @@ type StorageOptionsConf struct {
 	// Maximum TLS version that is supported.
 	// Options: ["1.0", "1.1", "1.2", "1.3"].
 	// Defaults to "1.3".
-	MaxVersion string `json:"max_version"`
+	MaxVersion string `json:"tls_max_version"`
```
Comment (titpetric) [inline config/config.go:157]: You'll have to change the field too, it ends up in a config `env` var, TYK_STORAGE_TLSMAXVERSION...

File: config/config.go [hunk 2]
```diff
@@ -424,10 +424,10 @@ type HttpServerOptionsConfig struct {
 	ServerName string `json:"server_name"`
 
 	// Minimum TLS version. Possible values: https://tyk.io/docs/basic-config-and-security/security/tls-and-ssl/#values-for-tls-versions
-	MinVersion uint16 `json:"min_version"`
+	TLSMinVersion uint16 `json:"min_version"`
```
Comment (titpetric) [inline config/config.go:427]: This may be breaking, pre-exists before storage, revert please

**Comment by buger** [comment]
**API tests result - postgres15-sha256 env: success** :white_check_mark: 
Branch used: refs/pull/6078/merge
Commit:  
Triggered by: pull_request (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/8065879059)

**Review by titpetric — APPROVED** [review]
LGTM

**Comment by buger** [comment]
**API tests result - postgres15-murmur64 env: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 6dac26dac0ac3aeadc29578bc90c57cce77031c9 [TT-11405] Updating JSON tags and field names for TLS max and min versions (#6078)

## **User description**


## Description
Updating JSON tags and field names for TLS max and min versions. 

https://tyktech.atlassian.net/browse/TT-11405


## Related Issue
https://tyktech.slack.com/archives/C02AS4BP5HN/p1709041100516449





## Motivation and Context



## How This Has Been Tested






## Screenshots (if appropriate)

## Types of changes



- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test
coverage to functionality)

## Checklist





- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning
why it's required
- [ ] I would like a code coverage CI quality gate exception and have
explained why


___

## **Type**
enhancement


___

## **Description**
- Updated JSON tags and field names related to TLS version configuration
across multiple files to improve clarity and consistency.
- Renamed `MinVersion` and `MaxVersion` to `TLSMinVersion` and
`TLSMaxVersion` in `HttpServerOptionsConfig` for better readability.
- Adjusted references in the TLS configuration setup in various parts of
the codebase to align with the new field names.
- Ensured TLS version consistency and updated the default TLS max
version to `VersionTLS12` in the gateway server configuration.


___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant
files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
<summary><strong>config.go</strong><dd><code>Update TLS version field
names and JSON tags</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

config/config.go
<li>Updated JSON tags for <code>MaxVersion</code> and
<code>MinVersion</code> to <code>tls_max_version</code> and
<br><code>tls_min_version</code> respectively in
<code>StorageOptionsConf</code>.<br> <li> Renamed fields
<code>MinVersion</code> and <code>MaxVersion</code> to
<code>TLSMinVersion</code> and <br><code>TLSMaxVersion</code> in
<code>HttpServerOptionsConfig</code>.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17">+4/-4</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>dashboard_register.go</strong><dd><code>Align TLS
version configuration references</code>&nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; </dd></summary>
<hr>

gateway/dashboard_register.go
<li>Updated references from <code>MinVersion</code> and
<code>MaxVersion</code> to <code>TLSMinVersion</code> and
<br><code>TLSMaxVersion</code> in the TLS configuration.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-f504c88b3d2fa3b56b74c252aab41a934156879ef1150d33714225749e6cc94c">+2/-2</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>proxy_muxer.go</strong><dd><code>Update TLS version
configuration in proxy muxer</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/proxy_muxer.go
<li>Adjusted TLS configuration to use the new <code>TLSMinVersion</code>
and <br><code>TLSMaxVersion</code> fields.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-89fb6731880400cb95ba8860c935a308de5f55aaa41aa2c76abf3ee4773d7a87">+2/-2</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>rpc_storage_handler.go</strong><dd><code>Update RPC
storage handler TLS version configuration</code>&nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; </dd></summary>
<hr>

gateway/rpc_storage_handler.go
<li>Updated RPC configuration to use <code>TLSMinVersion</code> and
<code>TLSMaxVersion</code> for <br>SSL version control.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-8875f75b602664c44b62b67a4da41d748124ad270573a44db4ec977ee5d68021">+2/-2</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>server.go</strong><dd><code>Enforce TLS version
consistency and update defaults</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; </dd></summary>
<hr>

gateway/server.go
<li>Ensured TLS version consistency and updated the default TLS max
<br>version to <code>VersionTLS12</code>.<br> <li> Updated checks and
assignments to use <code>TLSMinVersion</code> and
<code>TLSMaxVersion</code>.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-4652d1bf175a0be8f5e61ef7177c9666f23e077d8626b73ac9d13358fa8b525b">+4/-4</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools
and their descriptions
Triggered by: push (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/8096995450)

**Comment by buger** [comment]
**API tests result - mongo44-murmur64 env: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: 6dac26dac0ac3aeadc29578bc90c57cce77031c9 [TT-11405] Updating JSON tags and field names for TLS max and min versions (#6078)

## **User description**


## Description
Updating JSON tags and field names for TLS max and min versions. 

https://tyktech.atlassian.net/browse/TT-11405


## Related Issue
https://tyktech.slack.com/archives/C02AS4BP5HN/p1709041100516449





## Motivation and Context



## How This Has Been Tested






## Screenshots (if appropriate)

## Types of changes



- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test
coverage to functionality)

## Checklist





- [ ] I ensured that the documentation is up to date
- [ ] I explained why this PR updates go.mod in detail with reasoning
why it's required
- [ ] I would like a code coverage CI quality gate exception and have
explained why


___

## **Type**
enhancement


___

## **Description**
- Updated JSON tags and field names related to TLS version configuration
across multiple files to improve clarity and consistency.
- Renamed `MinVersion` and `MaxVersion` to `TLSMinVersion` and
`TLSMaxVersion` in `HttpServerOptionsConfig` for better readability.
- Adjusted references in the TLS configuration setup in various parts of
the codebase to align with the new field names.
- Ensured TLS version consistency and updated the default TLS max
version to `VersionTLS12` in the gateway server configuration.


___



## **Changes walkthrough**
<table><thead><tr><th></th><th align="left">Relevant
files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
<summary><strong>config.go</strong><dd><code>Update TLS version field
names and JSON tags</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

config/config.go
<li>Updated JSON tags for <code>MaxVersion</code> and
<code>MinVersion</code> to <code>tls_max_version</code> and
<br><code>tls_min_version</code> respectively in
<code>StorageOptionsConf</code>.<br> <li> Renamed fields
<code>MinVersion</code> and <code>MaxVersion</code> to
<code>TLSMinVersion</code> and <br><code>TLSMaxVersion</code> in
<code>HttpServerOptionsConfig</code>.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-fe44f09c4d5977b5f5eaea29170b6a0748819c9d02271746a20d81a5f3efca17">+4/-4</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>dashboard_register.go</strong><dd><code>Align TLS
version configuration references</code>&nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; </dd></summary>
<hr>

gateway/dashboard_register.go
<li>Updated references from <code>MinVersion</code> and
<code>MaxVersion</code> to <code>TLSMinVersion</code> and
<br><code>TLSMaxVersion</code> in the TLS configuration.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-f504c88b3d2fa3b56b74c252aab41a934156879ef1150d33714225749e6cc94c">+2/-2</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>proxy_muxer.go</strong><dd><code>Update TLS version
configuration in proxy muxer</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

gateway/proxy_muxer.go
<li>Adjusted TLS configuration to use the new <code>TLSMinVersion</code>
and <br><code>TLSMaxVersion</code> fields.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-89fb6731880400cb95ba8860c935a308de5f55aaa41aa2c76abf3ee4773d7a87">+2/-2</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>rpc_storage_handler.go</strong><dd><code>Update RPC
storage handler TLS version configuration</code>&nbsp; &nbsp; &nbsp;
&nbsp; &nbsp; </dd></summary>
<hr>

gateway/rpc_storage_handler.go
<li>Updated RPC configuration to use <code>TLSMinVersion</code> and
<code>TLSMaxVersion</code> for <br>SSL version control.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-8875f75b602664c44b62b67a4da41d748124ad270573a44db4ec977ee5d68021">+2/-2</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    

<tr>
  <td>
    <details>
<summary><strong>server.go</strong><dd><code>Enforce TLS version
consistency and update defaults</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
&nbsp; </dd></summary>
<hr>

gateway/server.go
<li>Ensured TLS version consistency and updated the default TLS max
<br>version to <code>VersionTLS12</code>.<br> <li> Updated checks and
assignments to use <code>TLSMinVersion</code> and
<code>TLSMaxVersion</code>.


</details>
    

  </td>
<td><a
href="https:/TykTechnologies/tyk/pull/6078/files#diff-4652d1bf175a0be8f5e61ef7177c9666f23e077d8626b73ac9d13358fa8b525b">+4/-4</a>&nbsp;
&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> ✨ **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools
and their descriptions
Triggered by: push (@mativm02)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/8096995450)

**Pull request merged**